clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -1875,21 +1883,15 @@ void FVComponentVariable::mergeDeclaration(FVComponentVariable *From,
 
 std::string
 FVComponentVariable::mkString(const EnvironmentMap &E) const {
-  std::string Str;
-  if (ExternalConstraint->anyChanges(E) && InternalConstraint->anyChanges(E))
-    Str = ExternalConstraint->mkString(E);
-  else {
-    Str = ExternalConstraint->getRewritableOriginalTy() +
-          ExternalConstraint->getName();
-    if (ExternalConstraint->anyChanges(E))
-      Str += " : itype(" + ExternalConstraint->mkString(E, false, true) + ")";
-  }
-  return Str;
+  return mkTypeStr(E, true) + mkItypeStr(E);
```

Comment:
nice simplification. Will it handle qualified types and anything else that was in the `getRewritableOriginalTy()`?